pkg/kubelet/dockershim/docker_image_linux.go

```diff
@@ -20,12 +20,58 @@ package dockershim
 
 import (
 	"context"
-	"fmt"
+	"path/filepath"
+	"os"
+	"time"
+
+	"k8s.io/klog"
 
 	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
 )
 
 // ImageFsInfo returns information of the filesystem that is used to store images.
 func (ds *dockerService) ImageFsInfo(_ context.Context, r *runtimeapi.ImageFsInfoRequest) (*runtimeapi.ImageFsInfoResponse, error) {
-	return nil, fmt.Errorf("not implemented")
+	info, err := ds.client.Info()
+	if err != nil {
+		klog.Errorf("Failed to get docker info: %v", err)
+		return nil, err
+	}
+
+	bytes, inodes, err := dirSize(filepath.Join(info.DockerRootDir, "image"))
+	if err != nil {
+		return nil, err
+	}
+
+	return &runtimeapi.ImageFsInfoResponse{
+		ImageFilesystems: []*runtimeapi.FilesystemUsage{
+			&runtimeapi.FilesystemUsage{
+				Timestamp: time.Now().Unix(),
+				FsId: &runtimeapi.FilesystemIdentifier{
+					Mountpoint: info.DockerRootDir,
+				},
+				UsedBytes: &runtimeapi.UInt64Value{
+					Value: uint64(bytes),
+				},
+				InodesUsed: &runtimeapi.UInt64Value{
```

Comment:
Are we concerned that this is including information not included in the same method in `docker_image_windows`?

Reply:
The difference should be fine. Currently kubelet doesn't use this value from the response in it's logic, so having it here and not in the windows version won't break anything. And I think we should aim to implement the CRI standard fully when possible.

Reply:
Agree

Reply:
Windows doesn't have inodes, and those shouldn't be considered as part of schedulable